hoomd/CommunicatorGPU.cc

```diff
@@ -1490,8 +1490,6 @@ void CommunicatorGPU::migrateParticles()
         // determine local particles that are to be sent to neighboring processors and fill send buffer
         uint3 mypos = m_decomposition->getGridPos();
 
-        /* We need some better heuristics to decide whether to take the GPU or CPU code path */
-        #if 0
```

Comment:
> Very interested to know how this performs.

Thanks, @vyasr!

Without detailed performance measurements I can just say it performs *decently*.  However, note that this is not the final iteration of this algorithm (more changes forthcoming in the next PR, as also noted above). One main change in the acceptance rule in this PR is to propose trial moves, and then to realize every temporary accept/reject decision per particle and re-check overlaps, *for each iteration of the acceptance kernel sequence*. This obviously represents an opportunity for optimization if we are able to pre-calculate the free energy change as a function of those decisions, rather than re-evaluating them several times (typically 3-4 times). As a consequence, this present implementation is a little slower than it could be. But this kind of logic was easier to implement in a first attempt with the two phase depletant insertion kernels.

In the next PR, the pre-calculation of overlaps will be (or, in fact, has is already been :-) reverted to the previous behavior of checking old and new configurations in one go, for improved performance. This also explains the change in the shift factors (`<< 2`) you correctly noted below, as those shifts were tracking the old and the new configuration of the neighboring particle.

Reply:
Got it, so some of this code involves known performance regressions to simplify the 1-phase to 2-phase conversion.

Reply:
> It appears that this code path was never safe before, but now the CPU fallback has been completely removed. What changed?

See below (comment on CommunicatorGPU.cu)